persistence-test/src/main/java/io/stargate/it/PersistenceTest.java

```diff
@@ -1964,6 +1966,42 @@ public void testRowDecorator() throws ExecutionException, InterruptedException {
     assertGt(last, first, dec1, dec2);
   }
 
+  @Test
+  public void testRowDecoratorComparableBytes() throws ExecutionException, InterruptedException {
+    setupCustomPagingData();
+
+    // Obtain partition keys in "ring" order
+    AbstractBound<?> selectAll =
+        dataStore
+            .queryBuilder()
+            .select()
+            .column("pk")
+            .column("val")
+            .from(keyspace, table)
+            .build()
+            .bind();
+
+    ResultSet rs1 = dataStore.execute(selectAll).get();
+    RowDecorator dec1 = rs1.makeRowDecorator();
+    List<Row> rows = rs1.rows();
+    if (backend.isDse()) {
+      List<String> allComparableByteStrings =
+          rows.stream()
+              .map(dec1::getComparableBytes)
+              .map(ByteComparable::fixedLength)
+              .map(x -> x.byteComparableAsString(ByteComparable.Version.DSE68))
+              .collect(Collectors.toList());
```

Comment:
You shouldn't depend on these DSE related interfaces? What's the issue with comparing bytes in unsigned order directly?

I know there is `UnsignedBytes` in Guava, this should have something we could use.. Or use something like this (take from `PureJavaOperations` class):

```
    @Override
    public int compare(ByteBuffer buffer1, ByteBuffer buffer2) {
      int end1 = buffer1.limit();
      int end2 = buffer2.limit();
      for (int i = buffer1.position(), j = buffer2.position(); i < end1 && j < end2; i++, j++) {
        int a = (buffer1.get(i) & 0xff);
        int b = (buffer2.get(j) & 0xff);
        if (a != b) {
          return a - b;
        }
      }
      return buffer1.remaining() - buffer2.remaining();
    }
```